src/test/groovy/pl/touk/service/ParkingRateServiceSpec.groovy

```diff
@@ -0,0 +1,102 @@
+package pl.touk.service
+
+import pl.touk.model.Driver
+import pl.touk.model.ParkingRate
+import pl.touk.repository.DriverRepository
+import pl.touk.repository.ParkingRateRepository
+import pl.touk.service.dto.NewDriverConverter
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static pl.touk.model.CurrencyType.PLN
+import static pl.touk.model.CustomerType.REGULAR
+import static pl.touk.model.CustomerType.VIP
+
+class ParkingRateServiceSpec extends Specification {
+
+    def vipRate = new ParkingRate(null, VIP, 0d, 2d, 1.2d, PLN)
+    def regularRate = new ParkingRate(null, REGULAR, 1d, 2d, 1.5d, PLN)
+    def rates = [vipRate, regularRate]
+
+    def driverRepository = Mock(DriverRepository)
+    def parkingRateRepository = Mock(ParkingRateRepository)
+
+    def service = new ParkingRateService(driverRepository, parkingRateRepository)
+
+    def setup() {
+        parkingRateRepository.findAll() >> rates
+        service.init()
+    }
+
+    @Unroll
+    def "should count fee for vip and regular: '#execNr'"() {
```

Comment:
Czy execNr jest potrzebne?

Reply:
Jest to tylko udogodnienie. Poszczególne przypadki testowe są numerowane. Jasno widać, który przypadek nie przeszedł.